src/meltano/core/plugin/base.py

```diff
@@ -327,8 +327,10 @@ def __init__(
         plugin_type: PluginType,
         name: str,
         namespace: str,
+        *,
         variant: str | None = None,
         variants: list | None = None,
```

Comment:
This is legacy stuff. I'll experiment a bit to see if it can be safely removed (after this PR is merged).